compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1088,7 +1088,19 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         errorTree(tree, em"cannot convert to type selection") // will never be printed due to fallback
     }
 
-    if (tree.qualifier.isType) {
+    def warnUnnecessaryNN(tree: Tree): Unit = {
+      if ctx.explicitNulls then {
```

Comment:
Do you agree we should check unnecessary nn even without explicit nulls?

Reply:
We could, but I didn't want to rock the boat too much before we fine-tune which corner cases should/shouldn't warn.